DeviceCleanup.ps1

```diff
@@ -0,0 +1,196 @@
+﻿<#
+    .SYNOPSIS
+    Generates a report of all devices in your tenant, including the last signed in date (if any) based on the last activity.
+    Optionally, it can remove devices if they have been inactive for a given threshold number of days by supplying the removeInactiveDevices switch
+
+    If the nonInteractive switch is supplied, the script will leverage Managed Identity (e.g. when running as an Azure Runbook) to log in to the Graph API. 
+    Assign the Device.Write.All permissions to the managed identity by using: https://gitlab.com/Lieben/assortedFunctions/-/blob/master/add-roleToManagedIdentity.ps1
+
+    If you want the script to send mail reports, also assign a value for the From, To addresses and assign the Mail.Send graph permission to the managed identity as per above instructions.
+
+    If the firstDisableDevices switch is also supplied, devices will not be deleted when the inactiveThresholdInDays is met, but disabled instead. Then after the 
+    disableDurationInDays threshold, they will be deleted, unless the device has already been inactive for longer than inactiveThresholdDays + disableDurationInDays
+
+    .NOTES
+    filename:   get-AzureAdInactiveDevices.ps1
+    author:     Jos Lieben / jos@lieben.nu
+    copyright:  Lieben Consultancy, free to (re)use, keep headers intact
+    disclaimer: https://www.lieben.nu/liebensraum/contact/#disclaimer-and-copyright
+    site:       https://www.lieben.nu
+    Created:    16/12/2021
+    Updated:    See Gitlab
+#>
+#Requires -Modules @{ ModuleName="Az.Accounts"; ModuleVersion="2.7.0" }, @{ ModuleName="Az.Resources"; ModuleVersion="5.1.0" }
+
+Param(
+    [Int]$inactiveDisableThresholdInDays = 90,
+    [Int]$inactiveDeleteThresholdInDays = 180,
+    [Switch]$removeInactiveDevices = $False,
+    [Switch]$disableInactiveDevices =$True,
+    [Switch]$nonInteractive=$True,
+    [String]$mailFrom ="jzahumesky@uniphar.ie", #this should not be a shared mailbox
+    [String[]]$mailTo ="jzahumesky@uniphar.ie"

```

Comment:
Can you remove the email address from params since it's a public repo? We can pass parameter values during invocation.